tests/integration/cqlengine/columns/test_validation.py

```diff
@@ -187,49 +184,173 @@ def test_varint_io(self):
         int2 = self.VarIntTest.objects(test_id=0).first()
         self.assertEqual(int1.bignum, int2.bignum)
 
+        with self.assertRaises(ValidationError):
+            self.VarIntTest.objects.create(test_id=0, bignum="not_a_number")
 
-class TestDate(BaseCassEngTestCase):
-    class DateTest(Model):
-
-        test_id = Integer(primary_key=True)
-        created_at = Date()
 
+class DataType():
     @classmethod
     def setUpClass(cls):
-        if PROTOCOL_VERSION < 4:
+        if PROTOCOL_VERSION < 4 or CASSANDRA_VERSION < "3.0":
             return
-        sync_table(cls.DateTest)
+
+        class DataTypeTest(Model):
+            test_id = Integer(primary_key=True)
+            class_param = cls.db_klass()
+
+        cls.model_class = DataTypeTest
+        sync_table(cls.model_class)
 
     @classmethod
     def tearDownClass(cls):
-        if PROTOCOL_VERSION < 4:
+        if PROTOCOL_VERSION < 4 or CASSANDRA_VERSION < "3.0":
             return
-        drop_table(cls.DateTest)
+        drop_table(cls.model_class)
 
     def setUp(self):
-        if PROTOCOL_VERSION < 4:
+        if PROTOCOL_VERSION < 4 or CASSANDRA_VERSION < "3.0":
             raise unittest.SkipTest("Protocol v4 datatypes require native protocol 4+, currently using: {0}".format(PROTOCOL_VERSION))
 
-    def test_date_io(self):
```

Comment:
Because it was merged in the test `test_date_io_using_datetime` , which I think I'll rename as it's supposed to be a generic class.